### PR TITLE
refs #159: massive refactoring of the repo based on previous configuration, plus updates.

### PR DESCRIPTION
**Pull request by stickgrinder**
@edodusi (/cc @paolomainardi) this commit restore all the functionalities the playbook used to have, such as non-destructive local build, exposition on port 80, not requiring local dependencies to work, etc.

Also it removes the dependency from Locke, building a local Raneto installation from node docker image.
All the updates @edodusi did with his last PR have been kept (maybe some more is needed, I'll double check), and a couple of patches that we used to apply, but have been accepted by Raneto maintainers, have been removed from the repo.

Also, the former directory structure has been preserved, giving Raneto an update custom config file.

It's a massive moving of stuff around, but the core of this is in Dockerfile, docker-compose and Makefile, plus the theme dependencies in package.json.

Let me know if this looks good. Thanks

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
 LABEL author="Paolo Pustorino <paolo.pustorino@sparkfabrik.com>"
 
-RUN apk add --no-cache --virtual .gyp npm make g++ py3-pip
+# Variables
+ENV INSTALL_DIR=/opt/raneto
+ENV PORT=80
 
 # Remove content folder
-RUN rm -rf /app/raneto/content
+RUN rm -rf content/
```
Comment (paolomainardi) [inline Dockerfile:9]: i don't think we need this line on the bare node image

File: Dockerfile [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
 RUN \
+  npm update && \
   npm install && \
   npm run build
 
-RUN apk del .gyp
+WORKDIR $INSTALL_DIR
+
+# Expose port 80
+EXPOSE 80
```
Comment (paolomainardi) [inline Dockerfile:33]: you can use `$PORT` as it has been defined on line 6

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
-WORKDIR /app/raneto/custom/themes/spark-playbook
+# Install raneto and deps
+WORKDIR $INSTALL_DIR
+RUN npm install
+
+WORKDIR $INSTALL_DIR/themes/spark-playbook
 
 RUN \
+  npm update && \
   npm install && \
```
[the remark this answers is not in the document]
Comment (stickgrinder) [reply Dockerfile:27]: TIL about `ci` command. Done

File: assets/images/fb_share_image.png
[the remark this answers is not in the document]
Comment (stickgrinder) [reply assets/images/fb_share_image.png:1]: I think the same but the problem should be addressed at a lower level, in Raneto itself. They are planning to make it by 0.19. Maybe we can work on a contribution.

Also, we can review this folder structure after we have everything in place in terms of build deps, ports etc.

File: Dockerfile (hunk 2 above)
Comment (stickgrinder) [reply Dockerfile:33]: Right! Done!

**Pull request merged**